File: daily/2026/2943_Maximize_Area_of_Square_Hole_in_Grid.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maximizeSquareHoleArea(int n, int m, vector<int>& hBars, vector<int>& vBars) {
        if (hBars.empty() || vBars.empty()) return 1;
        sort(hBars.begin(), hBars.end());
        sort(vBars.begin(), vBars.end());
        int hc = 2;
        int vc = 2;
        int hcurr = 2;
        for (int idx = 1; idx < hBars.size(); idx++){
            if (hBars[idx] == hBars[idx-1] + 1){
                hcurr++;
                hc = max(hc, hcurr);
            }
            else{
                hcurr = 2;
            }
        }
        int vcurr = 2;
        for (int idx = 1; idx < vBars.size(); idx++){
            if (vBars[idx] == vBars[idx-1] + 1){
                vcurr++;
                vc = max(vc, vcurr);
            }
            else{
                vcurr = 2;
            }
        }
        return min(hc, vc) * min(hc, vc);

    }
};

int main(){
    int n = 2;
    int m = 1;
    vector<int> hBars = vector{2,3};
    vector<int> vBars = {2};
    // int n = 1;
    // int m = 1;
    // vector<int> hBars = {2};
    // vector<int> vBars = {2};
    // int n = 2;
    // int m = 3;
    // vector<int> hBars = {2,3};
    // vector<int> vBars = {2,4};
    // int n = 3;
    // int m = 2;
    // vector<int> hBars = {3,2,4};
    // vector<int> vBars = {3,2};
    // int n = 10;
    // int m = 46;
    // vector<int> hBars = {10,5,6,8,11,3,2,7,4,9};
    // vector<int> vBars = {7,45,22,9,39,6};
    Solution S;
    cout << S.maximizeSquareHoleArea(n, m, hBars, vBars) << endl;
    return 0;
}

File: daily_python/2943_Maximize_Area_of_Square_Hole_in_Grid.py
import os
import sys
from typing import List

class Solution:
    def maximizeSquareHoleArea(self, n: int, m: int, hBars: List[int], vBars: List[int]) -> int:
        if not hBars or not vBars:
            return 1
        hBars.sort()
        vBars.sort()
        hc, vc = 2, 2
        maxhc = 2
        maxvc = 2
        for i in range(1, len(hBars)):
            if hBars[i] == hBars[i-1] + 1:
                hc += 1
                maxhc = max(maxhc, hc)
            else:
                hc = 2
        for i in range(1, len(vBars)):
            if vBars[i] == vBars[i-1] + 1:
                vc += 1
                maxvc = max(maxvc, vc)
            else:
                vc = 2
        return min(maxhc, maxvc) * min(maxhc, maxvc)

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            n = int(lines[0].split('=')[1].strip()[:-1])
            m = int(lines[1].split('=')[1].strip()[:-1])
            hBars = list(map(int, lines[2].split('=')[1].strip()[1:-2].split(',')))
            vBars = list(map(int, lines[3].split('=')[1].strip()[1:-2].split(',')))
            dataset.append((n, m, hBars, vBars))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for n, m, hBars, vBars in dataset:
        results.append(solution.maximizeSquareHoleArea(n, m, hBars, vBars))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')
